ambry-store/src/main/java/com/github/ambry/store/BootstrapController.java

```diff
@@ -0,0 +1,284 @@
+/**
+ * Copyright 2024 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import com.github.ambry.clustermap.ClusterParticipant;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.PartitionStateChangeListener;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.clustermap.StateModelListenerType;
+import com.github.ambry.clustermap.StateTransitionException;
+import com.github.ambry.config.ServerConfig;
+import com.github.ambry.config.ServerReplicationMode;
+import com.github.ambry.config.StoreConfig;
+import com.github.ambry.server.StoreManager;
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.regex.Pattern;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.clustermap.StateTransitionException.TransitionErrorCode.*;
+
+
+/**
+ * BootstrapController is responsible for managing the hydration protocol
+ * for Offline -> Bootstrap state transition of partitions.
+ */
+public class BootstrapController {
+  private boolean isRunning = false;
+  private final String BOOTSTRAP_IN_PROGRESS_FILE_NAME;
+  private final String FILECOPY_IN_PROGRESS_FILE_NAME;
+  private final Pattern allLogSegmentFilesPattern = Pattern.compile(".*_log.*");
```

Comment:
log segment file names would end up with _log, we don't need anything after.

Reply:
Incorporated.